test/TestFilePathEvaluation.py

```diff
@@ -0,0 +1,149 @@
+              debug:
+                msg: |
+                  Something...
+            """
+        ),
+    },
+}
+
+LAYOUT_INCLUDES = {
+    'directories': ['tasks', 'tasks/subtasks'],
+    'files': {
+        'main.yml': textwrap.dedent(
+            """\
+            ---
+            - hosts: target
+              gather_facts: false
+              tasks:
+                - name: from main import task 1
+                  include_tasks: tasks/task_1.yml
+            """
+        ),
+        'tasks/task_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 1 import task 2
+              include_tasks: tasks/task_2.yml
+            """
+        ),
+        'tasks/task_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 2 import subtask 1
+              include_tasks: tasks/subtasks/subtask_1.yml
+            """
+        ),
+        'tasks/subtasks/subtask_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 1 import subtask 2
+              include_tasks: tasks/subtasks/subtask_2.yml
+            """
+        ),
+        'tasks/subtasks/subtask_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 2 do something
+              debug:
+                msg: |
+                  Something...
+            """
+        ),
+    },
+}
+
+
+@pytest.fixture
+def create_root(tmp_path):
+    """Define root path for project."""
+    return str(tmp_path)
+
+
+@pytest.fixture
+def create_directory(tmp_path, relative_path):
+    """Create an empty directory in the project."""
+    directory = tmp_path / relative_path
+    return str(directory)
+
+
+@pytest.fixture
+def create_file(tmp_path, relative_path, content=""):
+    """Create a file in the project with specific content."""
+    file = tmp_path / relative_path
+    file.write_text(content)
+    return str(file)
+
+
+@pytest.mark.parametrize(
+    'ansible_project_layout',
+    (
+        pytest.param(LAYOUT_IMPORTS, id='using only import_tasks'),
+        pytest.param(LAYOUT_INCLUDES, id='using only include_tasks'),
+    ),
+)
+@pytest.mark.xfail(
+    reason='https://github.com/ansible-community/ansible-lint/issues/1446'
+)
+@pytest.mark.usefixtures('create_root', 'create_file', 'create_directory')
+def test_file_path_evaluation(default_rules_collection, ansible_project_layout):
+    """Test file path evaluation when using import_tasks / include_tasks in the project.
+
+    Usage of import_tasks / include_tasks may introduce false positive load-failure due
+    to incorrect file path evaluation.
+    """
+    root_dir_path = create_root()
+
+    for directory in ansible_project_layout['directories']:
+        create_directory(directory)
+
+    for file_path, file_content in ansible_project_layout['files'].items():
+        create_file(file_path, file_content)
+
+    result = Runner(root_dir_path, rules=default_rules_collection).run()
```

Comment:
Here's how to make this cleaner:

```suggestion
def test_file_path_evaluation(default_rules_collection, ansible_project_layout, tmp_path):
    """Test file path evaluation when using import_tasks / include_tasks in the project.

    Usage of import_tasks / include_tasks may introduce false-positive load-failure due
    to incorrect file path evaluation.
    """
    for file_path, file_content in ansible_project_layout['files'].items():
        full_path = tmp_path / file_path
        full_path.parent.mkdir(parents=True, exist_ok=True)
        full_path.write_text(file_content)

    result = Runner(str(tmp_path), rules=default_rules_collection).run()
```
(it also allows to simplify the layout object structure)